File: src/com/eagle/programmar/COBOL/Statements/COBOL_ReadStatement.java
// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Dec 6, 2010

package com.eagle.programmar.COBOL.Statements;

import com.eagle.programmar.COBOL.COBOL_AbstractStatement;
import com.eagle.programmar.COBOL.COBOL_Statement;
import com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference;
import com.eagle.programmar.COBOL.Terminals.COBOL_Keyword;
import com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice;
import com.eagle.tokens.TokenChooser;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;

public class COBOL_ReadStatement extends COBOL_AbstractStatement
{
	public @S(10) @DOC("rlpsread.htm") COBOL_Keyword READ = new COBOL_Keyword("READ");
	public @S(20) TokenList<COBOL_Identifier_Reference> files;
	public @S(30) TokenList<COBOL_ReadClause> entries;
	public @S(40) @OPT COBOL_Keyword ENDREAD = new COBOL_Keyword("END-READ");
	
	public static class COBOL_ReadClause extends TokenChooser
	{
		public @CHOICE COBOL_ReadInto into;
		public @CHOICE COBOL_ReadNext next;
		public @CHOICE COBOL_ReadAtEndAction atEnd;
		public @CHOICE COBOL_ReadIgnoreLock ignoreLock;
		public @CHOICE COBOL_ReadKey key;
		public @CHOICE COBOL_ReadInvalidKey invalid;
	}

	public static class COBOL_ReadInto extends TokenSequence
	{
		public @S(10) COBOL_Keyword INTO = new COBOL_Keyword("INTO");
		public @S(20) COBOL_Identifier_Reference var;
	}
	
	public static class COBOL_ReadNext extends TokenSequence
	{
		public @S(10) COBOL_Keyword NEXT = new COBOL_Keyword("NEXT");
		public @S(20) @OPT COBOL_Keyword RECORD = new COBOL_Keyword("RECORD");
	}
	
	public static class COBOL_ReadKey extends TokenSequence
	{
		public @S(10) @OPT COBOL_Keyword WITH = new COBOL_Keyword("WITH");
		public @S(20) COBOL_Keyword KEY = new COBOL_Keyword("KEY");
		public @S(30) COBOL_Keyword IS = new COBOL_Keyword("IS");
		public @S(40) COBOL_Identifier_Reference value;
	}

	public static class COBOL_ReadInvalidKey extends TokenSequence
	{
		public @S(10) @OPT COBOL_Keyword NOT = new COBOL_Keyword("NOT");
		public @S(20) COBOL_Keyword INVALID = new COBOL_Keyword("INVALID");
		public @S(30) COBOL_Keyword KEY = new COBOL_Keyword("KEY");
		public @S(40) TokenList<COBOL_Statement> statements;
	}

	public static class COBOL_ReadAtEndAction extends TokenSequence
	{
		public @S(10) COBOL_Keyword AT = new COBOL_Keyword("AT");
		public @S(20) COBOL_Keyword END = new COBOL_Keyword("END");
		public @S(30) TokenList<COBOL_Statement> endAction;
	}
	
	public static class COBOL_ReadIgnoreLock extends TokenSequence
	{
		public @S(10) @OPT COBOL_Keyword WITH = new COBOL_Keyword("WITH");
		public @S(20) @OPT COBOL_KeywordChoice IGNORE = new COBOL_KeywordChoice("IGNORE", "KEPT");
		public @S(30) COBOL_Keyword LOCK = new COBOL_Keyword("LOCK");
	}
}

File: csrc/com/eagle/programmar/COBOL/Statements/COBOL_ReadStatement.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Dec 6, 2010

namespace com.eagle.programmar.COBOL.Statements
{
	using COBOL_AbstractStatement = com.eagle.programmar.COBOL.COBOL_AbstractStatement;
	using COBOL_Statement = com.eagle.programmar.COBOL.COBOL_Statement;
	using COBOL_Identifier_Reference = com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference;
	using COBOL_Keyword = com.eagle.programmar.COBOL.Terminals.COBOL_Keyword;
	using COBOL_KeywordChoice = com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;

	public class COBOL_ReadStatement : COBOL_AbstractStatement
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @DOC("rlpsread.htm") com.eagle.programmar.COBOL.Terminals.COBOL_Keyword READ = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("READ");
		public @DOC("rlpsread.htm") COBOL_Keyword READ = new COBOL_Keyword("READ");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.tokens.TokenList<com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference> files;
		public TokenList<COBOL_Identifier_Reference> files;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) @OPT COBOL_Keyword RECORD = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("RECORD");
		public @OPT COBOL_Keyword RECORD = new COBOL_Keyword("RECORD");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) @OPT TokenList<COBOL_ReadClause> entries;
		public @OPT TokenList<COBOL_ReadClause> entries;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) @OPT COBOL_Keyword ENDREAD = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("END-READ");
		public @OPT COBOL_Keyword ENDREAD = new COBOL_Keyword("END-READ");

		public static class COBOL_ReadClause extends TokenChooser
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE COBOL_ReadInto XXinto;
			public COBOL_ReadInto XXinto;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE COBOL_ReadNext XXnext;
			public COBOL_ReadNext XXnext;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE COBOL_ReadAtEndAction XXatEnd;
			public COBOL_ReadAtEndAction XXatEnd;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE COBOL_ReadIgnoreLock XXignoreLock;
			public COBOL_ReadIgnoreLock XXignoreLock;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE COBOL_ReadKey XXkey;
			public COBOL_ReadKey XXkey;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE COBOL_ReadInvalidKey XXinvalid;
			public COBOL_ReadInvalidKey XXinvalid;
		}

		public static class COBOL_ReadInto extends TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword INTO = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("INTO");
			public COBOL_Keyword INTO = new COBOL_Keyword("INTO");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference var;
			public COBOL_Identifier_Reference var;
		}

		public static class COBOL_ReadNext extends TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword NEXT = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("NEXT");
			public COBOL_Keyword NEXT = new COBOL_Keyword("NEXT");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT COBOL_Keyword RECORD = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("RECORD");
			public @OPT COBOL_Keyword RECORD = new COBOL_Keyword("RECORD");
		}

		public static class COBOL_ReadKey extends TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @OPT COBOL_Keyword WITH = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("WITH");
			public @OPT COBOL_Keyword WITH = new COBOL_Keyword("WITH");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword KEY = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("KEY");
			public COBOL_Keyword KEY = new COBOL_Keyword("KEY");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword IS = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("IS");
			public COBOL_Keyword IS = new COBOL_Keyword("IS");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference value;
			public COBOL_Identifier_Reference value;
		}

		public static class COBOL_ReadInvalidKey extends TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @OPT COBOL_Keyword NOT = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("NOT");
			public @OPT COBOL_Keyword NOT = new COBOL_Keyword("NOT");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword INVALID = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("INVALID");
			public COBOL_Keyword INVALID = new COBOL_Keyword("INVALID");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword KEY = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("KEY");
			public COBOL_Keyword KEY = new COBOL_Keyword("KEY");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.tokens.TokenList<com.eagle.programmar.COBOL.COBOL_Statement> statements;
			public TokenList<COBOL_Statement> statements;
		}

		public static class COBOL_ReadAtEndAction extends TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword AT = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("AT");
			public COBOL_Keyword AT = new COBOL_Keyword("AT");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword END = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("END");
			public COBOL_Keyword END = new COBOL_Keyword("END");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.tokens.TokenList<com.eagle.programmar.COBOL.COBOL_Statement> endAction;
			public TokenList<COBOL_Statement> endAction;
		}

		public static class COBOL_ReadIgnoreLock extends TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @OPT COBOL_Keyword WITH = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("WITH");
			public @OPT COBOL_Keyword WITH = new COBOL_Keyword("WITH");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT COBOL_KeywordChoice IGNORE = new com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice("IGNORE", "KEPT");
			public @OPT COBOL_KeywordChoice IGNORE = new COBOL_KeywordChoice("IGNORE", "KEPT");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword LOCK = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("LOCK");
			public COBOL_Keyword LOCK = new COBOL_Keyword("LOCK");
		}
	}

}
